Feat: PCardSlider 컴포넌트 구현

thsk3 opened the pull request:
## 📝 Summary
<!-- PR에 대한 간략한 설명을 적어주세요 -->
공연 포스터 슬라이드 컴포넌트를 구현했습니다.

## ✨ Changes
<!-- 변경된 내용들을 상세히 나열해 주세요 -->
- react-slick 라이브러리를 설치했습니다.
- PCardSlider 컴포넌트를 추가했습니다.

## 🖼️ Screenshots
<!-- 필요한 경우 스크린샷을 첨부해 주세요 -->
<img width="434" alt="image" src="https://github.com/user-attachments/assets/308c69aa-c756-48bd-b498-ed5a406a1e16">
<img width="533" alt="image" src="https://github.com/user-attachments/assets/fbdd8fcb-a4e3-4062-aa3e-67985d7b0bb8">

## ✅ PR Checklist
<!-- 코드 리뷰 시 확인해야 할 사항들 -->
- [x] 하나의 목적을 가진 PR입니다.
- [x] 코딩 컨벤션과 스타일 가이드를 준수합니다. [📌Conventions](https://github.com/prgrms-fe-devcourse/NFE1-2-LocalStage/wiki/%F0%9F%93%8CConventions)
- [x] 불필요한 코드 중복이 없습니다.
- [x] 컴포넌트의 책임이 단일합니다.
- [ ] 리액트 훅을 올바르게 사용했습니다.
- [x] 민감한 정보를 포함하지 않았습니다.
- [x] 불필요한 콘솔 로그나 주석을 포함하지 않았습니다.
- [ ] 컴포넌트 key값에 고유한 값을 할당했습니다.

## 🔗 References
<!-- 관련된 이슈, PR, 링크 등을 첨부해 주세요 -->
- Issue: https://github.com/prgrms-fe-devcourse/NFE1-2-LocalStage/issues/79

## 💬 Comments
<!-- 추가적인 커멘트가 있다면 작성해 주세요 -->

juriurj committed:
Feat: PCardSlider 컴포넌트 구현
- react-slick 라이브러리 설치

shlee9999 commented on src/components/PCardSlider/index.tsx:
```diff
@@ -0,0 +1,37 @@
+  isRanked?: boolean;
+}
+
+export const PCardSlider = ({ pList, width = '600px', isRanked = false }: PCardSliderProps) => {
```
PCardSlider 스와이프 시 하나씩만 넘어가는 문제 해결하기

thsk3 replied:
해결했습니다!

shlee9999 commented on src/components/PCardSlider/index.tsx:
```diff
@@ -0,0 +1,37 @@
+              pInfo={perform}
+              width={`calc(${width} / 5.2)`}
+              {...(isRanked && { rank: index + 1 })}
+            ></PCard>
```
`<Pcard />` 형태로 바꾸기

shlee9999 commented on src/components/PCardSlider/index.tsx:
```diff
@@ -0,0 +1,37 @@
+              key={index}
+              pInfo={perform}
+              width={`calc(${width} / 5.2)`}
+              {...(isRanked && { rank: index + 1 })}
```
rank null 없애고 옵셔널 타입을 이용하기. PCardGallery, PCardSlider, PCardGrid 등

juriurj committed:
Feat: PCardSlider 슬라이더 스와이프 추가
- 커서로 스와이프 할 수 있게 함

juriurj committed:
Refactor: PCard 컴포넌트 props 수정
- rank 옵셔널 값으로 변경

juriurj committed:
Fix: PCardSlider의 요소 너비 수정
- PCardSlider의 너비를 %로 줬을 때, PCard의 너비도 %로 적용되는 것 수정
- 주어진 width의 형식을 확인하고 형식에 따라 PCard의 너비를 다르게 적용해줌

shlee9999 commented on src/components/PCardSlider/index.tsx:
```diff
@@ -0,0 +1,36 @@
+    <S.PCardSlider width={width}>
+      <Slider {...settings}>
+        {pList.map((perform, index) => {
+          return <PCard key={index} pInfo={perform} width={PCardWidth} {...(isRanked && { rank: index + 1 })} />;
```
`{...(isRanked && { rank: index + 1 })}` 말고 rank: ~~ 로 바꿔주세요!

shlee9999 commented on src/components/PCardSlider/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  isRanked?: boolean;
+}
+
+export const PCardSlider = ({ pList, width = '100%', isRanked }: PCardSliderProps) => {
```
isRanked 프로퍼티 없애고, pList에 rnum 프로퍼티를 옵셔널로 추가하여 이걸 이용해보기

juriurj committed:
Refactor: PCardSlider props 유연하게 적용
- PList나 Boxof 중 아무거나 줘도 잘 나오게 바꿈

shlee9999 commented on src/components/PCard/index.tsx:
```diff
@@ -4,19 +4,22 @@ import { Poster } from '@/components/Poster';
+export const PCard = ({ pInfo, width = '225px' }: PCardProps) => {
+  const poster = pInfo.rnum ? 'https://www.kopis.or.kr/' + pInfo.poster : pInfo.poster;
+  const place = pInfo.fcltynm ? pInfo.fcltynm : pInfo.prfnm;
+  const prf = pInfo.prfpd ? pInfo.prfpd : pInfo.prfpdfrom + '-' + pInfo.prfpdto;
```
이게 더 좋아보입니다!
 ```tsx
const prf = pInfo.prfpd ?? pInfo.prfpdfrom + '-' + pInfo.prfpdto;
```

shlee9999 committed:
Fix: PCard 관련 타입 및 구조 전부 수정
